VideoLocker/src/test/java/org/edx/mobile/test/http/ApiTests.java

```diff
@@ -333,6 +322,54 @@ public boolean apply(CourseComponent component) {
                 component = component.getParent();
             }
         }
+
+
     }
 
+    /**
+     * Verifies the parsing of course structure json to {@link CourseComponent} model class.
+     *
+     * @param component Parsed {@link CourseComponent} model.
+     */
+    private void verifyModelParsing(CourseComponent component) {
+        JSONObject jsonObj;
+        try {
+            jsonObj = new JSONObject(MockDataUtil.getMockResponse("get_course_structure"));
+            jsonObj = jsonObj.getJSONObject("blocks");
+            jsonObj = jsonObj.getJSONObject(component.getId());
+            assertNotNull(jsonObj);
+            // Not using the getDisplayName function below, because it returns a placeholder text
+            // when the display_name field's value is empty.
+            assertEquals(jsonObj.getString("display_name"), component.getInternalName());
+            assertEquals(jsonObj.getBoolean("graded"), component.isGraded());
+            assertEquals(jsonObj.getString("student_view_url"), component.getBlockUrl());
+            assertEquals(jsonObj.getBoolean("student_view_multi_device"), component.isMultiDevice());
+            assertEquals(jsonObj.getString("lms_web_url"), component.getWebUrl());
+
+            // Type specific validations
+            Gson gson = new Gson();
+            switch (component.getType()) {
+                case VIDEO: {
+                    String dataRawJson = jsonObj.getJSONObject("student_view_data").toString();
+                    assertTrue(component instanceof VideoBlockModel);
```

Comment:
Technically, per our current implementation, this assertion can only be performed if we assume that the unique properties of the video we present in the original JSON i.e "encoded_videos" and "transcripts".